### PR TITLE
Fix `custom-property-no-missing-var-function` false negatives for fallback values within `var()`

### LINKED ISSUE #8099: Fix `custom-property-no-missing-var-function` false negatives for fallback values within `var()`
### What minimal example or steps are needed to reproduce the bug?

```css
height: var(--primary-height, --fallback-height);
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "custom-property-no-missing-var-function": true
  }
}
```

### How did you run Stylelint?

```shell
stylelint “**/*.css” —config myconfig.json
```

### Which Stylelint-related dependencies are you using?

```json
{
        "stylelint": "^16.10.0",
        "stylelint-config-html": "^1.1.0",
        "stylelint-config-prettier-scss": "^1.0.0",
        "stylelint-config-standard": "^36.0.1",
        "stylelint-config-standard-scss": "^13.1.0",
        "stylelint-declaration-strict-value": "^1.10.6",
}
```

### What did you expect to happen?

Rule should have flagged the missing `var()` syntax on the fallback value.  The correct syntax would be:

```css
height: var(--primary-height, var(--fallback-height));
```

### What actually happened?

The test passed despite the missing `var()`

### Do you have a proposal to fix the bug?

_No response_

### PR DESCRIPTION
**Pull request by ryo-manba**
> Which issue, if any, is this issue related to?

Closes #8099 

> Is there anything in the PR that needs further explanation?

No, it's self-explanatory.

### CODE REVIEW

File: lib/rules/custom-property-no-missing-var-function/index.cjs [hunk 1]
```diff
@@ -65,29 +65,47 @@ const rule = (primary) => {
...
+		/**
+		 * @param {import('postcss-value-parser').Node} node
+		 * @param {import('postcss').Declaration} decl
+		 */
+		function check(node, decl) {
+			if (node.type === 'function') {
+				let args = node.nodes;
+
+				if (isVarFunction(node)) {
+					args = node.nodes.slice(1);
```
Comment (ryo-manba) [inline lib/rules/custom-property-no-missing-var-function/index.cjs:84]: Check fallback values when using `var` function.

File: lib/rules/custom-property-no-missing-var-function/index.mjs [hunk 2]
```diff
@@ -62,29 +62,45 @@ const rule = (primary) => {
...
 
-			valueParser(value).walk((node) => {
-				if (isVarFunction(node)) return false;
+			valueParser(value).nodes.forEach((childNode) => {
+				check(childNode, decl);
+			});
+		});
+
+		/**
+		 * @param {import('postcss-value-parser').Node} node
```
Comment (Mouvedia) [inline lib/rules/custom-property-no-missing-var-function/index.mjs:71]: nit

Maybe share the same typedef for this line and l108
i.e. `/** @typedef {import('postcss-value-parser').Node} Node  */`

```suggestion
		 * @param {Node} node
```

File: lib/rules/custom-property-no-missing-var-function/__tests__/index.mjs [hunk 3]
```diff
@@ -32,6 +32,10 @@ testRule({
 			code: 'a { color: var(--foo, red); }',
 			description: 'undeclared dashed-idents in var with fallback',
 		},
+		{
+			code: 'a { color: var(--foo, var(--bar)); }',
```
Comment (src-code) [inline lib/rules/custom-property-no-missing-var-function/__tests__/index.mjs:36]: It's possible for fallback values to also have fallbacks so would be good to have a test for that as well to ensure they're recursively tested, eg:

```css 
a { color: var(--foo, var(--bar, var(--foobar, #000))); }
```

File: lib/rules/custom-property-no-missing-var-function/index.mjs (hunk 2 above)
Comment (ryo-manba) [reply lib/rules/custom-property-no-missing-var-function/index.mjs]: Fixed it!
[chore: unify types](https://github.com/stylelint/stylelint/pull/8116/commits/c59d67c56dc847a58f661281563cfa2d5ae2c9af)

File: lib/rules/custom-property-no-missing-var-function/__tests__/index.mjs (hunk 3 above)
Comment (ryo-manba) [reply lib/rules/custom-property-no-missing-var-function/__tests__/index.mjs:36]: Thanks! I applied your suggestion.
[chore: add test for nested fallbacks](https://github.com/stylelint/stylelint/pull/8116/commits/b459db880c16a8525c7b3de41da85ec5c1af8b86)

**Comment by jeddy3** [comment]
@ryo-manba Thanks for working on this. When you have a moment, can you please resolve the conflict? Then we'll merge this.

**Comment by ryo-manba** [comment]
@jeddy3 I've fixed it!

**Review by jeddy3 — APPROVED** [review]
LGTM, thank you!

**Pull request merged**